src/libraries/System.Runtime/tests/System.Runtime.Tests/System/HalfTests.cs

```diff
@@ -410,6 +410,18 @@ public static IEnumerable<object[]> ExplicitConversion_ToSingle_TestData()
         [Theory]
         public static void ExplicitConversion_ToSingle(Half value, float expected) // Check the underlying bits for verifying NaNs
         {
+            if (PlatformDetection.IsRiscV64Process)
+            {
+                if (BitConverter.HalfToUInt16Bits(value) == 0b0_11111_1010101010)  // Positive Signalling NaN - RISC-V does not preserve payload
+                {
+                    expected = BitConverter.Int32BitsToSingle(0x7FC00000);
+                }
+                else if (BitConverter.HalfToUInt16Bits(value) == 0b1_11111_1010101010) // Negative Signalling NaN - RISC-V does not preserve payload
+                {
+                    expected = BitConverter.Int32BitsToSingle(unchecked((int)0xFFC00000));
+                }
```

Comment:
BTW, RISC-V doesn't preserve the NaN sign either. For Half->float conversion the sign is retained because it happens to be [ORed after the float op](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Half.cs#L1078-L1079). But that's a question of the expected behavior, i.e. whether Half->float conversion is expected to mimic e.g. double->float on a given architecture. Then again if the spec only quarantees some NaN is propagated, then I guess anything goes.

Reply:
@tannergooding could you please clarify what behavior we should expect for RISC-V? Should the half -> float conversion mimic float -> double?

Reply:
In this case I believe we should keep the behavior unaligned because changes most likely will cause performance degradation. @tannergooding could you please review this PR?

Reply:
> because changes most likely will cause performance degradation

This really comes down to a question of expected behavior, not performance. It doesn't matter how quickly code completes if it does the wrong thing, that just makes it incorrect.

This is very much a similar consideration to the expected behavior of `floating-point` to `integer` conversions where x86/x64 have their historical behavior (returning a sentinel value on overflow) and newer platforms have a different behavior (saturating) and where that behavior may be required by the platform specification (WASM).

It's a case where we are taking the breaking change and "performance hit" on x86/x64 to ensure the platforms are consistent and where newer hardware can minimize this imapct as it has instructions available that now do the "right thing".

-----------------

`NaN` handling isn't something we've made a formal decision on yet, but it is a case where there is a lot of code out there that does expect `NaN` propagation and where RISC-V is likely to hit issues by not propagating. This is likely going to particularly impact games and other high perf scenarios (such as SIMD or bit-twiddling hacks).

CC. @jkotas for additional input

Reply:
In general, RISC-V made number of design decisions that make more different from current mainstream platforms. I would wait and see how it is going to unfold. I think it is fine to special case the signaling NaN tests for now.